zsh-history-substring-search.zsh

```diff
@@ -246,7 +247,11 @@ _history-substring-search-begin() {
     # Escape and join query parts with wildcard character '*' as seperator
     # `(j:CHAR:)` join array to string with CHAR as seperator
     #
-    local search_pattern="*${(j:*:)_history_substring_search_query_parts[@]//(#m)[\][()|\\*?#<>~^]/\\$MATCH}*"
+    if [[ -n $HISTORY_SUBSTRING_SEARCH_PREFIX ]]; then
+        local search_pattern="${(j:*:)_history_substring_search_query_parts[@]//(#m)[\][()|\\*?#<>~^]/\\$MATCH}*"
+    else
+        local search_pattern="*${(j:*:)_history_substring_search_query_parts[@]//(#m)[\][()|\\*?#<>~^]/\\$MATCH}*"
```

Comment:
I think it might be cleaner (i.e. Don't Repeat Yourself) to keep the original definition of the base search pattern and strip the leading `*` for this feature.  Something like this:

```sh
test -n $HISTORY_SUBSTRING_SEARCH_PREFIX &&
local search_pattern=${search_pattern#?}
```

Reply:
It's a good idea. I have updated the code.